### PR TITLE
Adds support for chords in MusicXML importer

### LINKED ISSUE #667: Add support for chord diagrams in MusicXML
As reported [here](https://github.com/CoderLine/alphaTab/discussions/665) alphaTab does not support reading the individual frets and strings from the MusicXML harmony definition. We can use the information from the `frame` element to fill up all strings and frets of the chord accordingly:

https://w3c.github.io/musicxml/musicxml-reference/elements/frame/
[demo.xml.txt](https://github.com/CoderLine/alphaTab/files/7315535/demo.xml.txt)

### PR DESCRIPTION
**Pull request by coluzziandrea**
### Issues
Fixes #667 

image

### Checklist
- [x] I consent that this change becomes part of alphaTab under it's current or any future open source license
- [x] Changes are implemented
- [x] Existing builds tests pass
- [x] New tests were added

## Further details
- [ ] This is a breaking change
- [ ] This change will require update of the documentation/website

### CODE REVIEW

**Comment by coluzziandrea** [comment]
Hi @Danielku15, could you please take a look at this PR? 😊

**Review by Danielku15 — CHANGES_REQUESTED** [review]
Thanks for preparing this change and sorry for the delay. I currently have not much time to work on alphaTab. 

Can you maybe also take care that the C# variant compiles?

File: src/importer/MusicXmlImporter.ts [hunk 1]
```diff
@@ -555,9 +450,200 @@ export class MusicXmlImporter extends ScoreImporter {
...
+            chord.showDiagram = false;
+        }
         for (let staff of track.staves) {
             staff.addChord(this._currentChord, chord);
         }
+        chordsByIdForTrack.set(chordKey, chord);
+    }
+
+    private parseHarmonyRoot(xmlNode: XmlNode): string {
+        let rootStep: string | null = null;
```
[the remark this answers is not in the document]
Comment (coluzziandrea) [reply src/importer/MusicXmlImporter.ts:465]: you mean for cross-platform compiles?

[the remark this answers is not in the document]
Comment (Danielku15) [reply src/importer/MusicXmlImporter.ts:465]: On one hand yes. But also just from cleanliness in the code: We want to have an empty string anyhow, relying on the fact that null will become an empty string when concatenated is not so good.

**Comment by coluzziandrea** [comment]
@Danielku15 I've addressed comments, take a look when you have time please

**Comment by Danielku15** [comment]
@coluzziandrea Thanks a lot for your contribution. I took care of the remaining updates and cross compilation bits.

**Comment by goodgame365** [comment]
But after the update. the exported GP format file cannot be opened in Guitar Pro 8. When I try to open it, Guitar Pro 8 crashes.

**Pull request merged**